steps_file.js

```diff
@@ -32,16 +28,17 @@ module.exports = function () {
     async clearCart() {
       const elements = await this.grabNumberOfVisibleElements(basket);
       if (elements > 0) {
-        this.click(bucket);
-        let isElementPresent = await this.grabNumberOfVisibleElements(delButton) > 0;
-        while (isElementPresent) {
-            this.click(delButton); 
-            isElementPresent = await this.grabNumberOfVisibleElements(delButton) > 0; 
-        }
+          this.click(bucket);
+          let itemCount;
+          do {
+              itemCount = await this.grabNumberOfVisibleElements(delButton);
+              if (itemCount > 0) {
```

Comment:
this condition can be removed, you already check there is something in cart above, so at least one removal is required.
just move click before counting elements